allneeds.lua

```diff
@@ -0,0 +1,81 @@
+-- print sum of all needs
+--[====[
+
+allneeds
+========
+Show which needs are high for all dwarfs.
+Weight and Focus are divided by the number of dwarves that have that need.
+
+]====]
+local ENUM = {}
+
+ENUM[0]  = "Socialize"
+ENUM[1]  = "DrinkAlcohol"
+ENUM[2]  = "PrayOrMedidate"
+ENUM[3]  = "StayOccupied"
+ENUM[4]  = "BeCreative"
+ENUM[5]  = "Excitement"
+ENUM[6]  = "LearnSomething"
+ENUM[7]  = "BeWithFamily"
+ENUM[8]  = "BeWithFriends"
+ENUM[9]  = "HearEloquence"
+ENUM[10] = "UpholdTradition"
+ENUM[11] = "SelfExamination"
+ENUM[12] = "MakeMerry"
+ENUM[13] = "CraftObject"
+ENUM[14] = "MartialTraining"
+ENUM[15] = "PracticeSkill"
+ENUM[16] = "TakeItEasy"
+ENUM[17] = "MakeRomance"
+ENUM[18] = "SeeAnimal"
+ENUM[19] = "SeeGreatBeast"
+ENUM[20] = "AcquireObject"
+ENUM[21] = "EatGoodMeal"
+ENUM[22] = "Fight"
+ENUM[23] = "CauseTrouble"
+ENUM[24] = "Argue"
+ENUM[25] = "BeExtravagant"
+ENUM[26] = "Wander"
+ENUM[27] = "HelpSomebody"
+ENUM[28] = "ThinkAbstractly"
+ENUM[29] = "AdmireArt"
+
+
+local need = {}
+local n = 0
+for _, unit in ipairs(df.global.world.units.all) do
+    if not unit then
+      qerror('Unit not real.')
```

Comment:
This check isn't really necessary. For one thing, so many tools iterate over the units list without checking for null pointers that it's practically guaranteed that there won't be null pointers. And even if there are, since you're aborting the entire script at this point, you might as well just not have the check. Trying to access a field of `nil` will raise an error anyway, with more descriptive information, and won't crash DF (unlike a similar loop in C++, of which there are many in DFHack plugins). And `Unit not real` isn't really an appropriate description of the situation here. I'd just take out the check.